### PR TITLE
Convert Package to JS

### PR DESCRIPTION
**Pull request by maxbrunsfeld**

### CODE REVIEW

**Comment by 50Wliu** [comment]
@maxbrunsfeld any chance this can wait until #16100 is merged?

File: src/package.js [hunk 1]
```diff
@@ -0,0 +1,1106 @@
...
+  getActivationCommands () {
+    if (this.activationCommands) return this.activationCommands
+
+    this.activationCommands = {}
+
+    if (this.metadata.activationCommands) {
+      for (let selector in this.metadata.activationCommands) {
+        const commands = this.metadata.activationCommands[selector]
+        if (!this.activationCommands[selector]) this.activationCommands[selector] = []
+        if (_.isString(commands)) {
```
Comment (YurySolovyov) [inline src/package.js]: why not `typeof`?

File: src/package.js [hunk 2]
```diff
@@ -0,0 +1,1106 @@
...
+          this.activationCommands[selector].push(commands)
+        } else if (_.isArray(commands)) {
```
Comment (YurySolovyov) [inline src/package.js]: Why not `Array.isArray` ?

File: src/package.js [hunk 3]
```diff
@@ -0,0 +1,1106 @@
...
+    }
+
+    const incompatibleNativeModules = []
+    for (let nativeModulePath of this.getNativeModuleDependencyPaths()) {
+      try {
+        require(nativeModulePath)
+      } catch (error) {
+        let version
+        try {
+          ({version} = require(`${nativeModulePath}/package.json`))
```
Comment (YurySolovyov) [inline src/package.js]: are parens doing anything here?

File: src/package.js [hunk 4]
```diff
@@ -0,0 +1,1106 @@
...
+  disable () {
+    return this.config.pushAtKeyPath('core.disabledPackages', this.name)
+  }
+
+  isTheme () {
+    return this.metadata && this.metadata.theme
+  }
+
+  measure (key, fn) {
+    const startTime = Date.now()
```
Comment (YurySolovyov) [inline src/package.js]: `performance.now()` could be less expensive and more precise here

File: src/package.js (hunk 3 above)
Comment (50Wliu) [reply src/package.js]: Yes, you cannot have a bracket start a line because it interprets it as a block.

File: src/package.js [hunk 5]
```diff
@@ -0,0 +1,1106 @@
...
+      CompileCache.removeTranspilerConfigForPath(this.path)
+    }
+  }
+
+  loadKeymaps () {
+    if (this.bundledPackage && this.packageManager.packagesCache[this.name]) {
+      this.keymaps = []
+      for (const keymapPath in this.packageManager.packagesCache[this.name].keymaps) {
+        const keymapObject = this.packageManager.packagesCache[this.name].keymaps[keymapPath]
+        this.keymaps.push([`core:${keymapPath}`, keymapObject])
```
Comment (YurySolovyov) [inline src/package.js]: empty array + unconditional push seems like `.map` to me

**Comment by maxbrunsfeld** [comment]
> any chance this can wait until #16100 is merged?

Absolutely. This is low priority. Just something easy to do when I need a break from working on another problem.

File: src/package.js (hunk 5 above)
Comment (maxbrunsfeld) [reply src/package.js]: This is a loop over an object, so `for in` is convenient. An alternative would be `Object.keys(...).map`, though that would allocate an intermediate array of keys. Readability-wise, they seem pretty much equivalent.

**Pull request merged**